src/docfx/lib/schema/JsonSchemaTransformer.cs

```diff
@@ -74,11 +74,17 @@ public void PostValidate(bool isPartialBuild)
 
     public ExternalXref[] GetValidateExternalXrefs()
     {
-        return _xrefList.Value.Where(item => item.docsetName != null).GroupBy(item => item.xref.Value).Select(xrefGroup =>
+        return _xrefList.Value.Where(item => item.docsetName != null).GroupBy(item => new { item.xref.Value, item.propertyPath }).Select(xrefGroup =>
```

Comment:
Use tuple to avoid allocating an array.
```suggestion
        return _xrefList.Value.Where(item => item.docsetName != null).GroupBy(item => (item.xref.Value, item.propertyPath)).Select(xrefGroup =>
```